### PR TITLE
Switch to new JSX transform

### PR DESCRIPTION
**Pull request by mwiencek**
https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html

### CODE REVIEW

File: .eslintrc.yaml [hunk 1]
```diff
@@ -222,7 +222,7 @@ rules:
   react/no-will-update-set-state: error
   react/prefer-es6-class: off
   react/prop-types: off
-  react/react-in-jsx-scope: error
+  react/react-in-jsx-scope: off
```
Comment (reosarevok) [inline .eslintrc.yaml:225]: Maybe comment about why they're off, for future reference? We could also just remove them, not sure which one would be less confusing.

**Review by reosarevok — APPROVED** [review]
Seems reasonable. Navigated around a bit and it also seemed to work. Should we upgrade to React 17 next?

File: .eslintrc.yaml (hunk 1 above)
Comment (mwiencek) [reply .eslintrc.yaml:225]: A comment sounds good. I tend to prefer setting things to `off` just in case the defaults change.

**Comment by mwiencek** [comment]
> Should we upgrade to React 17 next?

Yes, once this is merged I'll work on that. :) (It was released right after I opened this PR.)

**Comment by mwiencek** [comment]
@brainzbot test this please

**Review by yvanzo — APPROVED** [review]

**Pull request merged**